NDTensors/ext/NDTensorsMetalExt/mul.jl

```diff
@@ -19,3 +19,21 @@ function LinearAlgebra.mul!(
   mul!(CM', BM', AM', α, β)
   return unexpose(CM)
 end
+
+## Fix issue in Metal.jl where it cannot distinguish Transpose{Reshape{Adjoint{CuArray}}}
+## as a CuArray and calls generic matmul
+function LinearAlgebra.mul!(
+  CM::Exposed{<:MtlArray},
+  AM::Exposed{<:MtlArray},
+  BM::Exposed{
+    <:MtlArray,
+    <:LinearAlgebra.Transpose{
+      <:Any,<:Base.ReshapedArray{<:Any,<:Any,<:LinearAlgebra.Adjoint}
+    },
+  },
+  α,
+  β,
+)
+  mul!(CM, AM, expose(transpose(copy(expose(parent(BM))))), α, β)
```

Comment:
@mtfishman would this be better split into two lines
```
B = copy(expose(parent(BM))
mul!(CM, AM, expose(transpose(B)), \alpha, \beta)
```

Reply:
Yeah, I think that looks better.

Reply:
Okay I made the change, thanks!